File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-4.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:  
        minn = float("inf")
        l = 0
        summ = 0
        for r in range(len(nums)):
            summ += nums[r]
            while summ >= target:
                W = r - l + 1
                minn = min(minn, W)
                summ -= nums[l]
                l += 1
            
        return 0 if minn == float("inf") else minn

File: Data Structures & Algorithms/combination-sum-iv/submission-4.go
func combinationSum4(nums []int, target int) int {
    dp := map[int]int{0: 1}

    for j := 1; j <= target; j++ {
        dp[j] = 0
        for _, num := range nums {
            dp[j] += dp[j-num]
        }
    }

    return dp[target]
}